testdata/testdata.go

```diff
@@ -105,121 +110,101 @@ func TestTable(name string) *schema.Table {
 	return sourceTable
 }
 
-func GenTestData(table *schema.Table) schema.CQTypes {
-	data := make(schema.CQTypes, len(table.Columns))
-	for i, c := range table.Columns {
-		switch c.Type {
-		case schema.TypeBool:
-			data[i] = &schema.Bool{
-				Bool:   true,
-				Status: schema.Present,
-			}
-		case schema.TypeInt:
-			data[i] = &schema.Int8{
-				Int:    1,
-				Status: schema.Present,
-			}
-		case schema.TypeFloat:
-			data[i] = &schema.Float8{
-				Float:  1.1,
-				Status: schema.Present,
-			}
-		case schema.TypeUUID:
-			uuidColumn := &schema.UUID{}
-			if err := uuidColumn.Set(uuid.NewString()); err != nil {
-				panic(err)
-			}
-			data[i] = uuidColumn
-		case schema.TypeString:
-			if c.Name == "text_with_null" {
-				data[i] = &schema.Text{
-					Str:    "AStringWith\x00NullBytes",
-					Status: schema.Present,
+func GenTestData(sc *arrow.Schema, sourceName string, syncTime time.Time, stableUUID uuid.UUID, count int) []arrow.Record {
+	var records []arrow.Record
+	for j := 0; j < count; j++ {
+		u := uuid.New()
+		if stableUUID != uuid.Nil {
+			u = stableUUID
+		}
+		bldr := array.NewRecordBuilder(memory.DefaultAllocator, sc)
+		for i, c := range sc.Fields() {
+			if arrow.TypeEqual(c.Type, arrow.FixedWidthTypes.Boolean) {
+				bldr.Field(i).(*array.BooleanBuilder).Append(true)
+			} else if arrow.TypeEqual(c.Type, arrow.PrimitiveTypes.Int64) {
+				bldr.Field(i).(*array.Int64Builder).Append(1)
+			} else if arrow.TypeEqual(c.Type, arrow.PrimitiveTypes.Float64) {
+				bldr.Field(i).(*array.Float64Builder).Append(1.1)
+			} else if arrow.TypeEqual(c.Type, types.ExtensionTypes.UUID) {
+				bldr.Field(i).(*types.UUIDBuilder).Append(u)
+			} else if arrow.TypeEqual(c.Type, arrow.BinaryTypes.String) {
+				if c.Name == schema.CqSourceNameColumn.Name {
+					bldr.Field(i).(*array.StringBuilder).AppendString(sourceName)
+				} else if c.Name == "text_with_null" {
+					bldr.Field(i).(*array.StringBuilder).AppendString("AStringWith\x00NullBytes")
+				} else {
+					bldr.Field(i).(*array.StringBuilder).AppendString("AString")
 				}
-			} else {
-				data[i] = &schema.Text{
-					Str:    "test",
-					Status: schema.Present,
+			} else if arrow.TypeEqual(c.Type, arrow.BinaryTypes.Binary) {
+				bldr.Field(i).(*array.BinaryBuilder).Append([]byte{1, 2, 3})
+			} else if arrow.TypeEqual(c.Type, arrow.ListOf(arrow.BinaryTypes.String)) {
+				if c.Name == "text_array_with_null" {
+					bldr.Field(i).(*array.ListBuilder).Append(true)
+					bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*array.StringBuilder).AppendString("test1")
```

Comment:
I think these ValueBuilders need to be `Released` to avoid memory leaks 🤔 Which makes me think, we may want to use `CheckedAllocator` in tests?